src/components/renderer/renderer.tsx

```diff
@@ -26,10 +27,22 @@ export default class Editor extends React.Component<Props> {
 
     const runtime = vega.parse(props.vegaSpec);
 
-    Editor.view = new vega.View(runtime)
-      .logLevel(vega.Warn)
-      .initialize(Editor.chart);
+    const loader = vega.loader();
+    const originalLoad = loader.load.bind(loader);
+    // Custom Loader
+    loader.load = async(url, options) => {
+      try {
+        return await originalLoad(url, {options, ...{baseURL: this.props.baseURL}});
```

Comment:
1. it should be ...options, right

Reply:
I guess you are thinking about options in options. We need to add baseURL directly in the options.
1. `...options` will append `baseURL` in options of options (nested object) while `options` will directly add baseURL in options. I tested it, working fine.
2. Adding an if-else block will work fine.
3. Ignored for now.

Reply:
So, if we go for `loader.load(url, {options: {baseURL: this.props.baseURL}})` it won't work.

Reply:
I'm confused. I thought the behavior of ...options and just options is flipped from how you describe it. I will test it when I'm.at a computer again.

Reply:
Yeah. This is correct. I was thinking it the wrong way, nevermind. I've updated the PR.

Reply:
Last question. Why override load and not http?

Reply:
Because `loader.http` method performs an HTTP request with a post-sanitization URL i.e we don't have baseURL option with this method. On the other hand, `loader.load` does all processing itself.
For more info check this https://github.com/vega/vega-loader/issues/7.